### PR TITLE
feat(executor_parameters): add resource_class and java_options as executor parameters

### PR DESCRIPTION
**Pull request by isaiahgrey93**
Allows the project `circle.yml` to configure the `resource_class` for each executor individually, and the `JAVA_OPTIONS` (passed as `java_options`) parameter passed to the android build process.

Example:
```yaml
version: 2.1
orbs:
  react-native: echobind/react-native
workflows:
  version: 2
  main:
    jobs:
      - react-native/node:
          name: node
          ...
      - react-native/android:
          custom_executor:
            name: react-native/android
            java_options: '-Xms2080m -Xmx3072m'
            resource_class: medium+
...
```

Example use case: https://circleci.com/gh/truecoach/truecoach-connect/527

### CODE REVIEW

File: src/orb.yml [hunk 1]
```diff
@@ -204,33 +204,45 @@ executors:
...
-
+    resource_class: <<parameters.resource_class>>
   android:
     parameters:
       version:
         type: string
         default: '27'
+      java_options:
+        type: string
+        default: '-Xms1048m -Xmx2048m'
```
Comment (cball) [inline src/orb.yml]:
```suggestion
        # sets the min and max memory for builds
        default: '-Xms1048m -Xmx2048m'
```

File: src/orb.yml [hunk 2]
```diff
@@ -204,33 +204,45 @@ executors:
       version:
         type: string
         default: '8'
+      resource_class:
+        type: string
+        default: medium
```
Comment (cball) [inline src/orb.yml]: is medium the lowest? Perhaps we should add a comment noting that this needs to be enabled by CircleCI w/ a link to the support page to do so?

File: src/orb.yml [hunk 3]
```diff
@@ -204,33 +204,45 @@ executors:
...
-
+    resource_class: <<parameters.resource_class>>
   ios:
     parameters:
       xcode_version:
         type: string
         default: '10.1.0'
+      resource_class:
+        type: string
+        default: medium
```
Comment (isaiahgrey93) [inline src/orb.yml]:
```suggestion
        # requires a support request to enable this parameter - default(medium)
        # https://circleci.com/docs/2.0/configuration-reference/#resource_class
        default: medium
```

File: src/orb.yml (hunk 2 above)
Comment (isaiahgrey93) [reply src/orb.yml]: Added a link to the first occurrence of the resource_class parameter.
https://github.com/echobind/circleci-react-native-orb/blob/6996c2f31b959a63d33054fd537dfb1909c4b1ab/src/orb.yml#L208-L209

**Pull request merged**